helsingborg-stad/app-mitt-helsingborg
# feat: added api for fetching application version status

## Explain the changes you’ve made
Added new service for fetching application version status, the status telling if the application version needs to be updated, new version exist or if application version is OK (i.e. up to date).

## Explain why these changes are made
We want to be able to tell the user if he or she needs to update the application, and adding the api for doing that is the first step.

## Explain your solution
Added a new file, `ApplicationVersionService.ts` which fetches the version status.

Added a new context, AppCompatibilityContext
- that consults ApplicationVersionService
- is used by MainNavigator to show alert if app needs update
- is used by AuthContext to force logout if app is incompatible
- is used by AuthContext to decide wether signing in is allowed (waits for compatible)

## How to test

1. Checkout this branch
2. Fire upp the simulator by running the command `yarn ios`
3. Make sure you have the version-api deployed and versionEnvs deployed 

## Tested environments

- [] Storybook on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.
